src/definition/rooms/IRoom.ts

```diff
@@ -23,4 +24,11 @@ export interface IRoom {
     customFields?: { [key: string]: any };
     parentRoom?: IRoom;
     livechatData?: { [key: string]: any };
+    sideBarIconSource?: {
```

Comment:
Considering this is an omnichannel property, so can we add this `ILivechatRoom` instead of the base room object?

Reply:
From what I understand, this property should not be exposed actually.

Reply:
> I mean, the idea of that prop is to not be exposed so others can modify (we didn't enforce this), currently, the apps-engine sets the value on room creation based on apps info (but this happens only for omnichannel rooms) since that prop should not exists on other room types

@KevLehman

Reply:
@ggazzo just created a PR with some modifcations on this PR https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/462#pullrequestreview-832976836

Reply:
I'm a bit confused with the implementation. It basically allows N number of channel icons (different for each room). That means an app would be able to set a different icon for every room it creates (which is a feature that can generate confusion to the end user as well)

From what I remember, an app would be able to use one icon and then all rooms created by the app would use that icon (if the app wanted it). Was this changed? 👀 

(Additionally, will we limit the ability to set custom room icons to only omnichannel rooms? I know this is what we need right now, but, with PR 462, the code validations would need to be removed if apps engine wanted to allow this in the future)

Reply:
we don't know if the same app can't have more than one integration, today we develop one by one, but it's up to the developer. The app will decide this when creating the room, this is a little less "plug and play", but at the same time more flexible, it involves less metadata and the behavior is also simpler. Today we are going to store the data in the room, otherwise we would need to store the information in the app, collect and evaluate this data, I don't think we have that prepared (I'm not an apps-engine expert, but I think it would take an infinite amount of work and a code more "coupled")

Today we are not interested in changing room icons to other types, we think this is a mess actually, but who knows, maybe in the future this 'source' field can live in all rooms not just omnichannel.

> I'm a bit confused with the implementation. It basically allows N number of channel icons (different for each room). That means an app would be able to set a different icon for every room it creates (which is a feature that can generate confusion to the end user as well)
> 
> From what I remember, an app would be able to use one icon and then all rooms created by the app would use that icon (if the app wanted it). Was this changed? 👀
> 
> (Additionally, will we limit the ability to set custom room icons to only omnichannel rooms? I know this is what we need right now, but, with PR 462, the code validations would need to be removed if apps engine wanted to allow this in the future)